engine/lib/filestore.php

```diff
@@ -741,6 +729,62 @@ function _elgg_filestore_move_icons($event, $type, $entity) {
 	}
 }
 
+/**
+ * Returns an array of uploaded file indices
+ * In case of a simple file input, will return [0]
+ * In case of a multiple file inputs with the same name, will produce ['pdf', 'doc']
+ * <code>
+ *   <input type="file" name="upload[pdf]" />
+ *   <input type="file" name="upload[doc]" />
+ * </code>
+ * In case of a file input with multiple attribute, will produce an array of numeric indices [0, 1, 2]
+ * <code>
+ *   <input type="file" name="upload[]" multiple />
+ * </code>
+ * In case of a missing input element, will return an empty array
+ *
+ * @param string $input_name Form input name
+ * @return array
+ */
+function elgg_get_uploaded_file_indices($input_name) {
+	return _elgg_services()->uploads->getUploadedFileIndices($input_name);
+}
+
+/**
+ * Returns uploaded file object or an array of uploaded file objects
+ *
+ * @param string $input_name Form input name
+ * @param string $index      Optional index of the file (if the input allows multiple files)
+ * @return UploadedFile|false
+ */
+function elgg_get_uploaded_file($input_name, $index = 0) {
+	return _elgg_services()->uploads->getUploadedFile($input_name, $index);
+}
+
+/**
+ * Returns contents of the uploaded file, or false if upload failed
+ *
+ * @param string $input_name Form input name
+ * @param string $index      Optional index of the file (if the input allows multiple files)
+ * @return string|false
+ */
+function elgg_read_uploaded_file($input_name, $index = 0) {
```

Comment:
Would prefer `elgg_get_uploaded_file_contents` but no big deal. Is this function necessary? We should avoid pulling file content into a string whenever we can, and instead use filesystem or stream APIs.